utils/windows-utils.ts

```diff
@@ -11,4 +14,106 @@ export function validatePackageID(id: string) {
+    validationErrors.push({ field: "name", error: "App name must be less than 256 characters" });
+  }
+
+  if (!options.version || options.version.trim().length === 0 && !options.classicPackage.version || options.classicPackage.version.trim().length === 0) {
+    validationErrors.push({ field: "version", error: "Must have an app version and a Classic Package version" });
+  }
+  else {
+    validateWindowsPackageVersion(options.version, options.classicPackage.version, validationErrors);
+  }
+
+  // Validating publisher options when we have publisher data
+  if (!options.publisher) {
+    validationErrors.push({ field: "publisher", error: "Publisher information must be supplied." });
+  } else {
+    // commonName and displayName are required, all other publisher fields are optional
+    const requiredPublisherFields: Array<keyof WindowsPublisherOptions> = [
+      "displayName",
+      "commonName"
+    ];
+
+    requiredPublisherFields
+      .filter(prop => !options.publisher![prop])
+      .forEach(() => validationErrors.push({ field: "publisher", error: `Publisher info must be specified` }));
+
+    // Ensure country code is 2 chars
+    if (options.publisher.country && options.publisher.country.length !== 2) {
+      validationErrors.push({ field: "publisher", error: "Publisher country code must be 2 letters" });
+    }
+  }
+
+  if (!options.url) {
+    validationErrors.push({ field: "url", error: "Start URL must be specified. If your start URL is the same as Host, you can use '/' as the start URL." });
+  } else {
+    const startUrlError = validateUrl(options.url, options.host);
+    if (startUrlError) {
+      validationErrors.push({ field: "url", error: "URL is invalid" });
+    }
+  }
+
+  return validationErrors;
+}
+
+function validateWindowsPackageVersion(version: string, classicVersion: string, validationErrors: { field: keyof WindowsPackageOptions | null, error: string }[]) {
+  console.log(version.trim(), version.trim().length);
+
+  /**
+   * Version must be in this form: "1.0.0"; a 3 segment version. 
+   * 4 segment versions (e.g. 1.0.0.0) are invalid due to Store restrictions; 
+   * the Store reserves the 4th segment for internal use.
+   * 
+   * versions in the 3 segment version should always only have 5 characters
+   */
+  if (version.trim().length !== 5) {
+    validationErrors.push({ field: "version", error: "Version must be in this form: 1.0.0; a 3 segment version." });
+  }
+
+  /**
+   * Version must be 1.0.0 or greater; Store doesn't support versions starting with zero.
+   */
+  const firstChar = version.trim().substr(0, 1);
+
+  if (firstChar && parseInt(firstChar) < 1) {
+    validationErrors.push({ field: "version", error: "Version must be 1.0.0 or greater; Store doesn't support versions starting with zero." });
+  }
+
+  /**
+   * classic package version should always be lower than app version
+   */
+
+  if (parseFloat(version.trim()) < parseFloat(classicVersion.trim())) {
```

Comment:
This won't work unfortunately.

- version is "2.6.1" 
- classic is "2.6.2"

This should be flagged as invalid (classic version must be lower than app version) but it will not be flagged as invalid.

Reply:
hmm ok cool, will fix this up too

Reply:
@JudahGabriel fixed this, turns out I didnt need parseFloat, JavaScript can compare accurately with them still as strings. parseFloat was losing the last decimal so that 2.6.2 and 2.6.1 looked the same.